src/consensus/tx_verify.cpp

```diff
@@ -169,3 +169,16 @@ bool CheckTransaction(const CTransaction& tx, bool fZerocoinActive, CValidationS
 
     return true;
 }
+
+bool ContextualCheckTransaction(const CTransactionRef& tx, CValidationState& state, const CChainParams& chainparams, int nHeight, bool fIBD)
+{
+    // Dispatch to Sapling validator
+    if (!SaplingValidation::ContextualCheckTransaction(*tx, state, chainparams, nHeight, true, fIBD)) {
```

Comment:
Not sure if this was on purpose but the `isMined` field is always true even when `ContextualCheckTransaction` is called from ATMP. Which will increase the misbehaving score for invalid mempool transactions.

-- If it was done in that way on purpose, i would prefer to divide it into its own commit and not here, plus remove the `isMined` member so the misbehaving score is the same for every call --

Reply:
No it wasn't on purpose, good catch. I think it's good to keep the different DOS scores.
Fixed in 706d12b1e0a6bc2f04fbab05ba977823d6c8baff